### ISSUE TITLE
&quot; is not escaped in the json output when included in an xml attribute

### ISSUE THREAD

**Issue by gavin-shaw** [shamblett/xml2json#23]
Hi xml2json!

Please check the below test:

```dart
import 'package:test/test.dart';
import 'package:xml2json/xml2json.dart';

void main() {
  test('xml2json escapes &quot; in the json output when used in an xml attribute', () {
    final xml = '<element att=" &quot;test&quot; "/>';

    final actual = (Xml2Json()..parse(xml)).toBadgerfish();
    final expected = '{"element": {"@att": " \\"test\\" "}}';

    expect(actual, equals(expected));
  });
}
```

It fails with:

```
          Expected: ... "@att": " \\"test\\" ...
            Actual: ... "@att": " "test" "}} ...
```

I expect that `&quot;test&quot;` in the xml attribute value should be escaped in the json output as `\"test\"`. Actually it is not escaped; `"test"`. So a subsequent jsonDecode() will fail at this node: ` "@att": " "test" "`.

I am using version:

```
  xml2json: ^4.3.0
```

**Comment by shamblett** [comment]
OK I'll have a look at this, thanks for the test.

**Comment by shamblett** [comment]
Applied fix, your test works OK now, packaged re-published at version 4.4.0 please re test.

**Comment by gavin-shaw** [comment]
It is working as expected, thank you for the quick response @shamblett !

**Comment by Time0o** [comment]
This issue is still present for some transformers, at least for OpenRally, can this be reopened?

**Comment by shamblett** [comment]
Yes, OpenRally was only recently added, I'll check the rest.

**Comment by shamblett** [comment]
OK I've updated OpenRally, this seems OK now. Parker is a different matter, I'm not sure this fix is applicable to Parker transforms, could you supply a fragment of XML showing the fault and what you expect Parker to transform it into, thanks.

**Comment by shamblett** [comment]
I'm happy with OpenRally so I've re published the package at version 6.2.0, I'll leave this open for a while for any sunsequent Parker commentary.

**Comment by hollingsworthd** [comment]
This is an issue with parerWithAttributes. The corrected code snippet in _Xml2JsonParkerWithAttrs is
```
  /// Analyze the attribute value in the node
  void _parseAttrs(dynamic node, dynamic obj) {
    node.attributes.forEach((attr) {
      obj!['"_${_Xml2JsonUtils.escapeTextForJson(attr.name.qualified)}"'] = '"${_Xml2JsonUtils.escapeTextForJson(attr.value)}"';
    });
  }
```

Many thanks for you hard work on this. I can make a PR if that's especially helpful, just let me know

**Comment by shamblett** [comment]
Hi, with your code added I've written the following unit test -
```
test(
      'xml2json escapes &quot; in the json output when used in an xml attribute - Parker with Attrs',
      () {
    final xml = '<element att=" &quot;test&quot; "/>';
    print(xml);
    final actual = (Xml2Json()..parse(xml)).toParkerWithAttrs();
    print(actual);
    final expected = '{"element": {"_att": " \\"test\\" ", "value": ""}}';
    expect(actual, equals(expected));
    final decoded = json.decode(actual);
    expect(decoded['element'], {'_att': ' "test" ', 'value': ''});
  });
```

Is this what you would expect?

**Comment by hollingsworthd** [comment]
Thanks! Yes that is as expected. I've also run this on real world data and it all seems to be working.

**Comment by shamblett** [comment]
Thanks for the confirmation. Package re released at version 6.2.4.
